rate_limit/provider.py

```diff
@@ -97,12 +103,51 @@ def get_local_rate_limits(self, scope, action, target_type_uri, **kwargs):
         :return: the local rate limit or -1 if not set
         """
         ttu_ratelimits = self.local_ratelimits.get(target_type_uri, [])
+        if not ttu_ratelimits:
+            ttu_ratelimits = self._get_wildcard_ratelimits(
+                self.local_ratelimits,
+                target_type_uri,
+            )
         for rl in ttu_ratelimits:
             ratelimit = rl.get('limit', None)
             if action == rl.get('action') and ratelimit:
                 return ratelimit
         return -1
 
+    def _get_wildcard_ratelimits(self, ratelimits, target_type_uri):
+        """
+        Get the target type URI rate limits from wildcard pattern.
+
+        :param target_type_uri: the target type URI of the request
+        :return: target type uri ratelimits if exists
+        """
+
+        ttu_ratelimits = []
+        pattern_list = [
+            lr_key for lr_key in ratelimits
+            if lr_key.endswith('*')
```

Comment:
Does this means that rule ending with "\*" is only allowed? Is "/foo/*/bar/*" allowed?

Reply:
Yes, only allowed for rules ending with "*". Discussed this with @galkindmitrii

Reply:
okay, in that case, I think it makes sense to validate the pattern somewhere once